### PR TITLE
time precision is returned in ColumnTypes()

### PR DESCRIPTION
**Pull request by mknister**
### Description
When there is a precision value for a time or timestamp column, a user is now able to retrieve that value using the ColumnTypes() function. A test created in driver_test.go confirms that it works.

### Checklist
- [ ] Code compiles correctly
- [ ] Run ``make fmt`` to fix inconsistent formats
- [ ] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by smtakeda** [comment]
@mknister can you fix the test failures?
```
driver_test.go:1982:2: this value of res is never used (SA4006)
driver_test.go:1982:7: this value of err is never used (SA4006)
gosnowflake.mak:29: recipe for target 'clint' failed
```
Also please create PR to prep-1.3.2

**Comment by smtakeda** [comment]
@mknister can you also rebase and push? There is a conflict in driver_test.go

File: Gopkg.lock [hunk 1]
```diff
@@ -15,7 +15,6 @@
   version = "1.1"
 
 [[projects]]
-  branch = "master"
```
Comment (smtakeda) [inline Gopkg.lock]: @mknister why did you remove?

File: query.go [hunk 2]
```diff
@@ -24,8 +24,8 @@ type execResponseRowType struct {
 	ByteLength int64  `json:"byteLength"`
 	Length     int64  `json:"length"`
 	Type       string `json:"type"`
-	Scale      int64  `json:"scale"`
 	Precision  int64  `json:"precision"`
+	Scale      int64  `json:"scale"`
```
Comment (smtakeda) [inline query.go]: Good catch!

File: driver_test.go [hunk 3]
```diff
@@ -1967,6 +1967,34 @@ func TestClientSessionKeepAliveParameter(t *testing.T) {
...
 
+func TestTimePrecision(t *testing.T) {
+	var db *sql.DB
+	var err error
+
+	if db, err = sql.Open("snowflake", dsn); err != nil {
+		t.Fatalf("failed to open db. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+	err = db.Ping()
```
Comment (smtakeda) [inline driver_test.go]: Ping is not required as you run db.Exec soon.

File: Gopkg.lock (hunk 1 above)
Comment (mknister) [reply Gopkg.lock]: @smtakeda this is something that Goland did automatically and I assumed it was on purpose. Should I revert this change?

File: driver_test.go (hunk 3 above)
Comment (mknister) [reply driver_test.go]: Ok, I will remove Ping.

File: Gopkg.lock (hunk 1 above)
Comment (smtakeda) [reply Gopkg.lock]: @ChTimTsubasa do you know about this line?

**Comment by smtakeda** [comment]
@mknister looks like still there is conflict. Please resolve it and update the change.

File: Gopkg.lock (hunk 1 above)
Comment (ChTimTsubasa) [reply Gopkg.lock]: This file is generated by dep. If dep changed it the we should respect its change.

File: driver_test.go [hunk 4]
```diff
@@ -1967,3 +1967,34 @@ func TestClientSessionKeepAliveParameter(t *testing.T) {
...
+
+	cols, _ := res.ColumnTypes()
+	pres, _, _ := cols[0].DecimalSize()
+	if pres != 5 {
+		t.Fatalf("Wrong value returned. Got %v instead of 5.", pres)
+	}
+}
+
+func init() {
+	if !flag.Parsed() {
```
Comment (ChTimTsubasa) [inline driver_test.go]: This is no longer supported in go 1.13. Please remove the whole init() function completely. See https://github.com/snowflakedb/gosnowflake/pull/267/files to look how the init should be performed.

File: driver_test.go [hunk 5]
```diff
@@ -1967,3 +1967,28 @@ func TestClientSessionKeepAliveParameter(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("error while executing query. err : %v", err)
+	}
+
+	cols, _ := res.ColumnTypes()
+	pres, _, _ := cols[0].DecimalSize()
+	if pres != 5 {
+		t.Fatalf("Wrong value returned. Got %v instead of 5.", pres)
+	}
+}
```
Comment (smtakeda) [inline driver_test.go]: Add a new line

File: driver_test.go (hunk 4 above)
Comment (smtakeda) [reply driver_test.go]: @ChTimTsubasa This change is no longer valid after rebase.

File: rows.go [hunk 6]
```diff
@@ -106,6 +106,10 @@ func (rows *snowflakeRows) ColumnTypePrecisionScale(index int) (precision, scale
 	switch rows.RowType[index].Type {
 	case "fixed":
 		return rows.RowType[index].Precision, rows.RowType[index].Scale, true
+	case "time":
+		return rows.RowType[index].Scale, 0, true
```
Comment (ChTimTsubasa) [inline rows.go]: We are setting the precision as the scale and the scale as 0, which is a little confusing to me. Let's hold the change for now and confirm with the customer what is the value he will be looking at.
